### PR TITLE
Add streaming JSON parser architecture

### PR DESCRIPTION
**Pull request by omerbenamram**
## Summary

This PR introduces a new streaming JSON parser architecture that bypasses intermediate XML model construction and serde_json::Value building, writing JSON directly to the output buffer.

## Key Changes

- **New streaming JSON parser** (`JsonStreamOutput`) that writes JSON directly without building intermediate structures
- **CLI option** `--json-parser` to choose between `legacy` (tree-based) and `streaming` (default) implementations
- **New API method** `records_json_stream()` that returns `SerializedEvtxRecord<String>` using the streaming path
- **Comprehensive test suite** (`test_full_samples_streaming.rs`) that validates streaming parser correctness and equivalence to legacy parser
- **Flamegraph improvements** with better tagging and output organization

## Performance Benefits

The streaming parser avoids:
- Building full `XmlModel` structures
- Creating intermediate `serde_json::Value` objects
- Multiple allocations and data transformations

## Testing

- All existing sample tests pass with streaming parser
- New equivalence tests ensure streaming output matches legacy parser output
- Tests cover both regular and `separate_json_attributes` modes

## Backward Compatibility

- Legacy parser remains available via `--json-parser legacy`
- Default behavior uses streaming parser for better performance
- All existing APIs continue to work unchanged

---

> [!NOTE]
> Introduces a streaming JSON parser selectable via --json-parser, adds records_json_stream API, comparison tool and tests, plus profiling helpers and CI/workflow updates.
> 
> - **Core/Library**:
>   - Add `JsonStreamOutput` for direct JSON streaming without `XmlModel`/`serde_json::Value` construction.
>   - Add `EvtxParser::records_json_stream()` and `EvtxRecord::into_json_stream()`.
>   - Implement streaming token path in `binxml/assemble.rs` (`parse_tokens_streaming`, helpers) and refactor template expansion off `Cow`.
>   - Export `JsonStreamOutput` in `lib.rs`.
> - **CLI**:
>   - Add `--json-parser` flag (`legacy`|`streaming`, default `streaming`) in `evtx_dump` and plumb through.
> - **New Binary/Tests**:
>   - Add `src/bin/compare_streaming_legacy.rs` to validate streaming vs legacy outputs; Make target `compare-streaming-legacy`.
>   - Add extensive streaming parity tests across samples and aggregation/duplicates scenarios.
> - **Profiling/Tooling**:
>   - Add `Makefile` and `scripts/flamegraph_prod.sh` with TAG-aware outputs; update docs to pass `TAG` and use tagged profile filenames.
>   - Ignore profiling/benchmark artifacts in `.gitignore`.
> - **CI/Workflows**:
>   - Update actions to latest (`checkout@v4`, `dtolnay/rust-toolchain@stable`), simplify test run; update Pages deploy workflow to new toolchain action.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9c220ee7ecc05c4085374da3a127f33ee9cb965c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by omerbenamram** [comment]
## Performance Benchmark Results

Comparing `master` vs `feature/json-streaming-parser-architecture` on `samples/security_big_sample.evtx`:

### Results

| Version | Mean Time | Std Dev | Range |
|---------|-----------|---------|-------|
| **master** (legacy) | 852.2 ms ± 53.0 ms | | 808.1 ms … 984.2 ms |
| **feature branch (streaming)** | **495.8 ms ± 26.7 ms** | | 472.8 ms … 595.3 ms |
| **feature branch (legacy)** | 863.7 ms ± 36.3 ms | | 826.9 ms … 944.4 ms |

### Summary

✅ **Streaming parser is 1.72× faster than master** (852.2 ms → 495.8 ms)
✅ **Streaming parser is 1.74× faster than legacy parser** on feature branch
✅ **Legacy parser performance maintained** (863.7 ms vs 852.2 ms master, ~1.4% slower, within noise)

### Benchmark Details

- **Command**: `evtx_dump -t 1 -o json samples/security_big_sample.evtx`
- **Warmup**: 10 runs
- **Runs**: 20
- **Features**: `fast-alloc` enabled
- **Sample**: `security_big_sample.evtx`

The streaming parser achieves a **~42% speedup** by avoiding intermediate XML model construction and `serde_json::Value` building, writing JSON directly to the output buffer.

**Comment by omerbenamram** [comment]
This obviously needs some cleanup, but a cool improvement (almost 2x)!

File: src/json_stream_output.rs [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+                        };
+                        if !first_field {
+                            self.write_bytes(b",")?;
+                        }
+                        serde_json::to_writer(self.writer_mut(), "#text")
+                            .map_err(SerializationError::from)?;
+                        self.write_bytes(b":")?;
+
+                        self.data_owner_depth = Some(owner_depth);
+                        self.data_array_started = false;
```
[the remark this answers is not in the document]
Comment (omerbenamram) [reply src/json_stream_output.rs]: good bot!
